code/modules/clothing/glasses/tajblind.dm

```diff
@@ -0,0 +1,196 @@
+			These current designs are adapted from recreations of the ancient veils, created by the Alchemists Guild. Technologically advanced and created to help Tajara adapt to life in the larger galactic community, they have systems built-in that allow them to have holographic huds, as well as corrective technology to help Tajaran overcome their genetic tritanopia colour blindness. <br>\
+			Availability on the wider market is highly restricted as a result of their cultural importance, as well as the patent held by the Alchemists Guild, and the lenses are very hard to reverse engineer. Popular theories suggest this as a result of the unique materials available on Adhomai, or the inability to recreate the light conditions of the Tajara homeworld."
+
+/obj/item/clothing/glasses/hud/tajblind/meson
+	name = "\improper Tajaran engineering meson veil"
+	icon_state = "tajblind_engi"
+	item_state = "tajblind_engi"
+
+/obj/item/clothing/glasses/hud/tajblind/meson/Initialize()
+	..()
+	desc += "<br><span class='info'>It has an optical meson scanner integrated into it.</span>"
+
+/obj/item/clothing/glasses/hud/tajblind/meson/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot == SLOT_HUD_GLASSES)
+		ADD_TRAIT(user, TRAIT_MESON_VISION, "meson_glasses[UID()]")
+
+/obj/item/clothing/glasses/hud/tajblind/meson/dropped(mob/user)
+	. = ..()
+	if(user)
+		REMOVE_TRAIT(user, TRAIT_MESON_VISION, "meson_glasses[UID()]")
+
+/obj/item/clothing/glasses/hud/tajblind/meson/cargo
+	name = "\improper Tajaran mining meson veil"
+	icon_state = "tajblind_cargo"
+	item_state = "tajblind_cargo"
+
+/obj/item/clothing/glasses/hud/tajblind/sci
+	name = "\improper Tajaran scientific veil"
+	icon_state = "tajblind_sci"
+	item_state = "tajblind_sci"
+	scan_reagents = 1
+	actions_types = list(/datum/action/item_action/toggle, /datum/action/item_action/toggle_research_scanner)
+
+/obj/item/clothing/glasses/hud/tajblind/sci/Initialize()
+	..()
+	desc += "<br><span class='info'>It has science goggles integrated into it.</span>"
+
+/obj/item/clothing/glasses/hud/tajblind/sci/item_action_slot_check(slot)
+	if(slot == SLOT_HUD_GLASSES)
+		return TRUE
+
+/obj/item/clothing/glasses/hud/tajblind/med
+	name = "\improper Tajaran medical veil"
+	icon_state = "tajblind_med"
+	item_state = "tajblind_med"
+	hud_types = DATA_HUD_MEDICAL_ADVANCED
+	examine_extensions = list(EXAMINE_HUD_MEDICAL_READ)
+
+/obj/item/clothing/glasses/hud/tajblind/med/Initialize()
+	..()
+	desc += "<br><span class='info'>It has a health HUD integrated into it.</span>"
+
+/obj/item/clothing/glasses/hud/tajblind/sec
+	name = "\improper Tajaran security veil"
+	icon_state = "tajblind_sec"
+	item_state = "tajblind_sec"
+	var/global/list/jobs[0]
```

Comment:
We generally don't use/define globals like this, what is this even used for?

Reply:
(We also don't define lists like this, we do `var/list/jobs = list()` )

Reply:
So, an amusing thing. I copied that code straight from current `/obj/item/clothing/glasses/hud/security` and since it worked, I completely forgot to actually check if the code I copied even makes sense. That one is definitely on me.
That likely empty list still exists in `hudglasses.dm`, yes.